models/auth/source.go

```diff
@@ -234,7 +234,7 @@ func CreateSource(ctx context.Context, source *Source) error {
 	err = registerableSource.RegisterSource()
 	if err != nil {
 		// remove the AuthSource in case of errors while registering configuration
-		if _, err := db.GetEngine(ctx).Delete(source); err != nil {
+		if _, err := db.GetEngine(ctx).Delete(&Source{ID: source.ID}); err != nil {
```

Comment:
```suggestion
		if _, err := db.GetEngine(ctx).ID(source.ID).Delete(new(Source)); err != nil {
```

Reply:
I feel `.ID` is a better idea too. But to be honest, I seldom use `new`. Is there a particular reason to use `new(Source)` instead of `&Source{}` for better readness? IMO, the compiler will translate them into the same code so there won't be a performance issue.